File: backend/app/categories/schemas/category.py
# app/schemas/category.py
from pydantic import BaseModel
from typing import Optional, List

class CategoryBase(BaseModel):
    name: str
    slug: str
    parent_id: Optional[int] = None
    is_active: bool = True

class CategoryCreate(CategoryBase):
    pass

class CategoryUpdate(BaseModel):
    name: Optional[str] = None
    slug: Optional[str] = None
    parent_id: Optional[int] = None
    is_active: Optional[bool] = None

class CategoryOut(CategoryBase):
    id: int

    class Config:
        from_attributes = True

File: frontend/src/types/category.ts
export type Category = {
  id: number;
  name: string;
  slug: string;
  description?: string | null;
  is_active: boolean;
};

export type CategoryCreatePayload = {
  name: string;
  slug: string;
  description?: string;
  is_active: boolean;
};
